blmath/geometry/primitives/plane.py

```diff
@@ -210,25 +210,22 @@ def polyline_xsection(self, polyline):
         return intersection_points
 
     def line_xsection(self, pt, ray):
-        pt = np.asarray(pt).ravel()
-        ray = np.asarray(ray).ravel()
-        assert len(pt) == 3
-        assert len(ray) == 3
+        return self._line_xsection(np.asarray(pt).ravel(), np.asarray(ray).ravel())
+
+    def _line_xsection(self, pt, ray):
         denom = np.dot(ray, self.normal)
         if denom == 0:
             return None # parallel, either coplanar or non-intersecting
         p = np.dot(self.reference_point - pt, self.normal) / denom
         return p * ray + pt
 
     def line_segment_xsection(self, a, b):
-        a = np.asarray(a).ravel()
-        b = np.asarray(b).ravel()
-        assert len(a) == 3
-        assert len(b) == 3
+        return self._line_segment_xsection(np.asarray(a).ravel(), np.asarray(b).ravel())
 
-        pt = self.line_xsection(a, b-a)
+    def _line_segment_xsection(self, a, b):
+        pt = self._line_xsection(a, b-a)
         if pt is not None:
-            if np.any(pt < np.min((a, b), axis=0)) or np.any(pt > np.max((a, b), axis=0)):
+            if any(np.logical_and(pt > a, pt > b)) or any(np.logical_and(pt < a, pt < b)):
```

Comment:
@algrs I get that we want things to be readable, but it's definitely more efficient to vectorize the calls to these. It appears to save a few seconds to use:

```
    def line_xsections(self, pts, rays):
        denoms = np.dot(rays, self.normal)
        denoms[denoms == 0] = np.nan
        p = np.dot(self.reference_point - pts, self.normal) / denoms
        return np.vstack([p, p, p]).T * rays + pts

    def line_segment_xsections(self, a, b):
        pts = self.line_xsections(a, b-a)
        pt_is_valid = ~np.any(np.isnan(pts), axis=1)
        valid_pts = pts[pt_is_valid]
        pt_is_out_of_bounds = np.logical_or(np.any(np.logical_and(pts > a, pts > b), axis=1), 
                                            np.any(np.logical_and(pts < a, pts < b), axis=1))
        valid_pts[pt_is_out_of_bounds, :] = np.nan
        pts[pt_is_valid] = valid_pts
        return pts
```
and then down below instead of the for loop over `es` you have

```
        vals = self._line_segment_xsections(m.v[es[:,0]], m.v[es[:,1]])
        val_is_valid = ~np.any(np.isnan(vals), axis=1)
        valid_vals = vals[val_is_valid]
        valid_es = es[val_is_valid]
        for val, e in zip (valid_vals, valid_es):
            if not intersection_map.contains(e[0], e[1]):
                intersection_map.add(e[0], e[1], val)
```

I'll try to come up with something that's a bit clearer